chore: update project dependencies and adjust Webpack configuration

pankaj28843 opened the pull request:
- Update project dependencies to latest versions for improved security and performance
- Modify .gitignore to exclude package-lock.json
- Create .nvmrc file to specify Node.js version for the project
- Adjust Webpack devServer static path in webpack.config.js

I tested this on the latest macOS, Windows 11, and Ubuntu 22.04

pankaj28843 committed:
chore: update project dependencies and adjust Webpack configuration

- Update project dependencies to latest versions for improved security and performance
- Modify .gitignore to exclude package-lock.json
- Create .nvmrc file to specify Node.js version for the project
- Adjust Webpack devServer static path in webpack.config.js

miltonhultgren commented on .nvmrc:
```diff
@@ -0,0 +1 @@
+v21.6
```
I would use the LTS version instead.

pankaj28843 replied:
Me too, I added this to align with what's in package.json. Will update

miltonhultgren reviewed (approved):
Looks good to me, consider using the LTS version in the `.nvmrc` instead

pankaj28843 committed:
fix: downgrade Node.js version in .nvmrc and package.json

- Downgrade Node.js version from v21.6 to v20.11.1 in .nvmrc file for compatibility issues
- Update node engine version in package.json to match the .nvmrc version

pankaj28843 commented:
Can someone with write access please merge this PR? 🙇🏻‍♂️

miltonhultgren commented:
@ahmagdy Has I think, I don't 🤷🏼

ahmagdy commented:
@pankaj28843 @miltonhultgren, you both have access now. Let me know if it doesn't work

pankaj28843 commented:
Thank you @ahmagdy, I was able to merge. 🙌🏻
